projects/plugins/boost/app/modules/image-guide/Image_Guide_Proxy.php

```diff
@@ -18,28 +18,27 @@ public static function init() {
 	public static function handle_proxy() {
 		// Verify valid nonce.
 		if ( empty( $_POST['nonce'] ) || ! wp_verify_nonce( sanitize_key( $_POST['nonce'] ), self::NONCE_ACTION ) ) {
-			die( 'bad nonce' );
+			wp_send_json_error( 'bad nonce', 400 );
 		}
 
 		// Make sure currently logged in as admin.
 		if ( ! current_user_can( 'manage_options' ) ) {
-			die( 'not admin' );
+			wp_send_json_error( 'not admin', 400 );
 		}
 
 		// Validate URL and fetch.
 		$proxy_url = filter_var( wp_unslash( isset( $_POST['proxy_url'] ) ? $_POST['proxy_url'] : null ), FILTER_VALIDATE_URL );
 		if ( ! wp_http_validate_url( $proxy_url ) ) {
-			die( 'Invalid URL' );
+			wp_send_json_error( 'Invalid URL', 400 );
 		}
 
 		$response = wp_remote_get( $proxy_url );
 		if ( is_wp_error( $response ) ) {
-			die( 'error' );
+			wp_send_json_error( 'error', 400 );
 		}
 
-		header( 'Content-type: text/html' );
-		echo (int) $response['headers']['content-length'];
-
+		header( 'Content-type: application/json' );
```

Comment:
You don't need to exclusively send the headers. `wp_send_json` does that for [you and also includes the charset](https://github.com/WordPress/wordpress-develop/blob/6.2/src/wp-includes/functions.php#L4412-L4417).

Reply:
Thanks, I did not know that!